syntax = "proto3";

package proto;

option csharp_namespace = "Costasdev.Busurbano.Backend.Types";

message Epsg25829 {
    double x = 1;
    double y = 2;
}

message StopArrivals {
    message ScheduledArrival {
        string service_id = 1;
        string trip_id = 2;
        string line = 3;
        string route = 4;
        string shape_id = 5;
        double shape_dist_traveled = 6;

        uint32 stop_sequence = 11;
        repeated string next_streets = 12;

        string starting_code = 21;
        string starting_name = 22;
        string starting_time = 23;

        string calling_time = 33;
        uint32 calling_ssm = 34;

        string terminus_code = 41;
        string terminus_name = 42;
        string terminus_time = 43;

        // Shape ID of the previous trip when the bus comes from another trip that ends at the starting point
        string previous_trip_shape_id = 51;
    }

    string stop_id = 1;

    Epsg25829 location = 3;

    repeated ScheduledArrival arrivals = 5;
}


message Shape {
    string shape_id = 1;

    repeated Epsg25829 points = 3;
}
